numpy/core/src/_simd/_simd_inc.h.src

```diff
@@ -147,22 +147,40 @@ typedef struct
 {
     PyObject_HEAD
     // vector type id
-    simd_data_type type;
+    simd_data_type dtype;
     // vector data, aligned for safe casting
     npyv_lanetype_u8 NPY_DECL_ALIGNED(NPY_SIMD_WIDTH) data[NPY_SIMD_WIDTH];
-} simd_vector;
+} simd_vector_obj;
```

Comment:
Would really encourage `simd_vector_Object`, because the point is to match `PyObject`, `PyTypeObject`, `PyArrayObject`, `...`. I suppose arguably a `Py` prefix would be good too.

Reply:
`simd_vector_Object` or `simd_vector_PyObject` looks weird if it's really necessary then we have to
change all functions names and prefix `simd_` to fit the python style.
for example:
  - `simd_` -> `PySIMD_`
  - `simd_vector_obj` -> `PySIMDVectorObject`
  - `simd_data` -> `PySIMDData`
  - `simd_data_type` -> `PySIMDDataType`

I feel like it would be an unnecessary burden and honestly, I don't like the Python API code style too long,
and strange to the C world.

@mattip, what do you think?

Reply:
To be clear, the renames i'm advocating for are:

* `simd__vector_type` &rarr; `PySIMDVectorType`
* `simd_vector_obj` &rarr; `PySIMDVectorObject`
* `simd_vector_...` &rarr; `PySIMDVector_...`

There is less need to match the python naming scheme when writing pure C code (`simd_data`, `simd_data_type` etc are fine). As soon as you're writing an api for a python object though, it makes a lot of sense to match the convention.

Reply:
> when writing pure C code

Do you mean when Python C API isn't involved?
If your answer is yes then I have to rename the following functions too:
```C
simd_scalar_from_obj
simd_scalar_to_obj
simd_sequence_from_obj
simd_sequence_fill_obj
simd_sequence_to_obj
simd_vectorx_from_obj
simd_vectorx_to_obj
simd_arg_from_obj
simd_arg_to_obj
simd_arg_converter
```

> As soon as you're writing an api for a python object though, it makes a lot of sense to match the convention.

Yes but SIMD is an internal module for testing NPYV intrinsics, and it doesn't expose any functions, so there's
no need to follow their code style.

Reply:
It is not about exposure, it is about using a consistent coding style across NumPy, as much as possible.

Reply:
Alright, I will strictly follow the NumPy/Python style.

Reply:
@mattip, I only renamed the following private functions and types into camel style as @eric-wieser requsted,
```C
simd__vector_type → PySIMDVectorType
simd_vector_obj → PySIMDVectorObject
simd_vector_to_obj → PySIMDVector_FromData
simd_vector_from_obj → PySIMDVector_AsData
simd_vector_register → PySIMDVectorType_Init
````

because I'm not sure what I suppose to do since according to [C style guide](https://numpy.org/doc/stable/dev/style_guide.html#naming-conventions)

> Private functions should be lower case with underscores, for example: array_real_get. Single leading underscores should not be used, but some current function names violate that rule due to historical accident. Those functions should be renamed at some point.